game-core/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java

```diff
@@ -37,7 +38,7 @@
   private int number = 0;
   private boolean allied = false;
   private boolean enemy = false;
-  private String bonusType = null;
+  private Tuple<Integer, String> bonusType = null;
```

Comment:
I think we're trying to phase out `Tuple` (@RoiEXLab (?))

This can be replaced pretty readily by a value object, eg:

```
@Value
private static class {
  @Nonnull Integer ...
  @Nonnull String ...
}
```

We get benefit that the values become labelled. It's not immediately clear what the 'string' or even the 'int' represent. Following up with naming/abstractions, should this variable still be called 'bonusType?'

Reply:
Ok. I'm pretty neutral on tuple vs a small nested class. Though there is a pretty good amount of tuple use so probably should consider adding a note about that in coding standards and it would probably be better to have a pure refactoring PR to address at least the majority of it to push for consistency.

Reply:
One of the cases where commentary in PR did not make it into the code. Agree it should be documented, probably ideally as deprecated.

It's been an ongoing process to remove, note for example "Triplet" is removed. There is far too much code to remove in one go for it to be done in one PR. "Pair" objects are common, there are many in the code base, it's consistency within a class, package, project, and we never had that consistency in package or project.

For some time we have not been adding new dependencies.